backend/de.metas.business/src/main/java/de/metas/forex/ForexContractService.java

```diff
@@ -174,4 +185,54 @@ public ImmutableSet<ForexContractId> getContractIdsEligibleToAllocateOrder(
 						.displayNameSearchTerm(displayNameSearchTerm)
 						.build());
 	}
+
+	@NonNull
+	private BooleanWithReason isOrderEligibleToUnallocate(
+			@NonNull final OrderId orderId,
+			@NonNull final ForexContractId contractId)
+	{
+		final BooleanWithReason isOrderEligible = checkOrderCompleted(orderId);
+		if (isOrderEligible.isFalse())
+		{
+			return isOrderEligible;
+		}
+
+		final boolean hasAllocations = hasAllocations(orderId);
+		if (!hasAllocations)
+		{
+			return BooleanWithReason.falseBecause("Order has no FEC allocations!");
+		}
+
+		return invoiceBL.hasInvoiceWithForexContract(orderId, contractId)
+				? BooleanWithReason.trueBecause("Order has invoices created with Forex contract ID!")
+				: BooleanWithReason.FALSE;
+	}
+
+	private void validateEligibleToUnallocate(
+			@NonNull final OrderId orderId,
+			@NonNull final ForexContractId contractId)
+	{
+		final BooleanWithReason isOrderEligibleToUnallocate = isOrderEligibleToUnallocate(orderId, contractId);
```

Comment:
minor: rename `isOrderEligibleToUnallocate` to `orderEligibleToUnallocate` beause IMHO orderEligibleToUnallocate.isFalse() sounds better